### PR TITLE
Implemented find node

### PR DESCRIPTION
**Pull request by asta987**

### CODE REVIEW

File: kademlia-node/src/kademlia/message_handler.go [hunk 1]
```diff
@@ -39,6 +39,21 @@ func (messageHandler *MessageHandler) HandleMessage(rawMessage []byte) []byte {
 
 		return bytes
 
+	case FIND_NODE:
+		var findN Find_Node
+
+		json.Unmarshal(rawMessage, &findN)
+		fmt.Println(findN.ID)
```
Comment (MrDweller) [inline kademlia-node/src/kademlia/message_handler.go:46]: Remove ID print

File: kademlia-node/src/kademlia/message_handler.go [hunk 2]
```diff
@@ -39,6 +39,21 @@ func (messageHandler *MessageHandler) HandleMessage(rawMessage []byte) []byte {
...
+
+		fmt.Println(findN.FromAddress + " wants to find your k closest nodes.")
+		closestKNodesList := messageHandler.kademliaNode.RoutingTable.FindClosestContacts(findN.ID, 3)
```
Comment (MrDweller) [inline kademlia-node/src/kademlia/message_handler.go:49]: Make 3, variable in kademliaNode

File: kademlia-node/src/kademlia/message.go [hunk 3]
```diff
@@ -54,3 +55,20 @@ func NewAckPingMessage(fromAddress string) Pong {
 		fromAddress,
 	}
 }
+
+type Find_Node struct {
```
Comment (arianfiftyone) [inline kademlia-node/src/kademlia/message.go:59]: Pascalcase on struct, i.e. `FindNode`

File: kademlia-node/src/kademlia/network.go [hunk 4]
```diff
@@ -116,8 +116,22 @@ func (network *Network) SendPingMessage(contact *Contact) bool {
 
 }
 
-func (network *Network) SendFindContactMessage(contact *Contact) {
-	// TODO
+func (network *Network) SendFindContactMessage(contact *Contact) bool {
```
Comment (MrDweller) [inline kademlia-node/src/kademlia/network.go:119]: Return response, i.e list of contacts and not just a bool

**Review by MrDweller — CHANGES_REQUESTED** [review]

File: kademlia-node/src/kademlia/message_handler.go (hunk 1 above)
Comment (arianfiftyone) [inline kademlia-node/src/kademlia/message_handler.go:46]: Unecessary print?

**Review by arianfiftyone — CHANGES_REQUESTED** [review]
Explicit requests. Otherwise we probably need more gracious error handling throughout the repository.

**Review by arianfiftyone — APPROVED** [review]

**Review by MrDweller — APPROVED** [review]

**Pull request merged**